Feature/parallel chart

Huongg opened the pull request:
## Description

Resolves #1154 

Single run for parallel coordinates chart using dummy data

## Development notes

Features and function that include in this PR:
- Matching the design for the chart in single run mode

WIP:
<img width="1790" alt="Screenshot 2022-11-14 at 20 08 18" src="https://user-images.githubusercontent.com/32060364/201668154-7db68654-9516-4c24-8bc0-2bb0af63899c.png">


Design:
<img width="814" alt="Screenshot 2022-11-14 at 15 13 32" src="https://user-images.githubusercontent.com/32060364/201608620-230f8ed2-a797-494b-8d28-1308be0132bb.png">

- Hover interaction between the run list and the chart itself. For example, when you hover over a line in the chart, or from the run list, it will highlight the line and labels of that run, however, all the related label values in each axis won’t be highlighted.
- Hover interaction always works for the Y-axis (metrics name)

- When the run hovers, its line always comes on top of any other lines. 

What does not include in this PR: the Tooltip, the chart in comparison mode, light theme


## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes


<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1152"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

Huongg commented on src/components/experiment-tracking/parallel-coordinates/parallel-coordinates.js:
```diff
@@ -0,0 +1,215 @@
+          width,
+        }}
+      >
+        {/* {graphKeys.map((graph) => (
```
I can't find a way to make this Yaxis work in the component yet, still WIP

comym commented:
Hi,

Here are my comments:

1 - Background colour is wrong. It should be Slate-900 all throughout and not have multiple divisions of colours as you see here. The top tab bar should blend with the background of the whole page.

![Screenshot 2022-11-14 at 15 55 34](https://user-images.githubusercontent.com/55446576/201705619-dfc39202-ce5c-4eee-bac4-064eb6e7f9cb.png)


2 - Is it my impression or are we using 2px lines in the graph? It all looks chunkier. Not sure, just wanted to double-check. We should use 1px lines.


![Screenshot 2022-11-14 at 15 57 01](https://user-images.githubusercontent.com/55446576/201706046-1eac319a-8937-4023-ad37-dcee7353ce63.png)

3 - There's some sort of tick/dash mark on top of the first metric label.

![Screenshot 2022-11-14 at 15 57 32](https://user-images.githubusercontent.com/55446576/201706236-b67406dd-7342-4ad8-91a9-8069ac5adfb6.png)


4 - The bottom margin is too narrow. We should give it more room. The metrics axis should shorten and the bottom margin increase in size.

![Screenshot 2022-11-14 at 16 00 13](https://user-images.githubusercontent.com/55446576/201707346-ee9ae7e5-e684-42db-bfad-698f5c393910.png)


5 - The lines are all crossing on top of the metric axis. This should not happen. Axis, value labels and their complementary lines should always be on top. Lines should be at the back and label values and tick marks on top.

![Screenshot 2022-11-14 at 16 04 25](https://user-images.githubusercontent.com/55446576/201708296-742ccdc6-b863-4bfe-a6c5-eb4613b6d874.png)

![Screenshot 2022-11-14 at 16 04 42](https://user-images.githubusercontent.com/55446576/201708299-9b305f05-25e4-4b2a-a38c-ab48d67a4cc9.png)

![Screenshot 2022-11-14 at 16 16 12](https://user-images.githubusercontent.com/55446576/201710840-7bab1309-d2cf-4ed1-a49f-ec9f0816e2ee.png)

![Screenshot 2022-11-14 at 16 16 18](https://user-images.githubusercontent.com/55446576/201710844-0610156c-f708-4d05-80a1-7e60ae554e39.png)



6 - One of my main concerns is, it's still very fiddly to hover over any of the lines in the graph. 

7 - When hovering on the card, the lines are flickering when moving the mouse inside the respective cards. This should not happen.

![Screen Recording 2022-11-14 at 16 09 54](https://user-images.githubusercontent.com/55446576/201709407-eb216fce-1fe1-4d95-aab6-c3635bdb152c.gif)


8 - Top and bottom labels of the axis range should not be this bright and they should not be semi-bold.

![Screenshot 2022-11-14 at 16 13 55](https://user-images.githubusercontent.com/55446576/201709993-56f2d1e9-734e-4297-89d6-66c0e7a9ccf4.png)


9 - Font sizes for selected run values and non-selected ones are different. It should be the same.

![Screenshot 2022-11-14 at 16 22 28](https://user-images.githubusercontent.com/55446576/201712332-fd0b280f-8df8-4595-9bdc-6a227313b6c9.png)


10 - On hover, run labels show to be inconsistent. Sometimes highlighted, in a different colour, or non-existent.

![Screenshot 2022-11-14 at 16 26 43](https://user-images.githubusercontent.com/55446576/201713269-111b9a7b-5b76-4b14-93e2-47b53bd8d8ad.png)


11 - On hover, neither the highlighted metric axis nor the run lines should be white FFFFFF. 

![Screenshot 2022-11-14 at 16 29 04](https://user-images.githubusercontent.com/55446576/201713780-248346e0-13a2-4be1-8fd2-2005184f0f49.png)



Please use the provided designs for reference. All colours and styles are there. 
We'll be working on the light version for that too. 

Thanks

Huongg commented:
hey @comym thanks for taking a look at this PR and thanks for the feedback. Very helpful as I've been looking at this everyday and it's hard to spot these issues. 

I belive I've addressed your comments with more details as below:

**1. Background colour is wrong**. Yes noted this is wrong but it's something defined from the code base a long time ago. Will double check with Tynan about this issue. 

2. Line width is now set back to the default value which is 1px lines.

3. Those ticks on the top of the chart have now been removed

4. I've increased the margin for the chart bottom

5. The lines are now rendered at the bottom of the chart, so that the labels and other lines can stay on the top

6. I see your concern about the behaviour of hovering over the line. We haven't found the perfect solution yet, I'd suggest including this item to the next iteration. What do you think @tynandebold ?

7. This is now fixed, when hovering over the run card the lines wont be disappeared/appeared again

8. I've doubled checked and updated all the colours according to the figma files. Let me know if you spot anything else? 

9. Font sizes are now also updated with the figma file. However, on the selected run, the values do come with a font-weight: 600, which will make it looks slightly different from other values. Is it correct?

comym commented:
Hi @Huongg 

Thanks for the comments and iteration. My feedback is:

1 - Lines are still crossing on top of the axis and label values.

![Screenshot 2022-11-15 at 11 17 57](https://user-images.githubusercontent.com/55446576/201906947-568dbf95-a50b-4125-9314-a25e954e5a9e.png)


2 - The top and bottom range value labels seem to be too bright. Make sure they are Grey-300 in their original state.

![Screenshot 2022-11-15 at 11 22 17](https://user-images.githubusercontent.com/55446576/201907718-9773e6e8-1688-4c19-945b-f75945eb77d7.png)

3 - The top and bottom range value labels seem to have a different size/weight than the other labels in the range.

![Screenshot 2022-11-15 at 11 24 49](https://user-images.githubusercontent.com/55446576/201908150-b8d12cf3-531d-4369-b178-0f16a62f558d.png)

4 - The bottom margin is not right. The way I see it here, it now leaves a giant gap at the bottom. The margins should be proportional to the viewport and always show a balanced layout this way. At the moment the margins are varying and affecting the layout depending on the screen size.

![Screenshot 2022-11-15 at 11 32 50](https://user-images.githubusercontent.com/55446576/201909910-d20348de-6a3d-4247-9888-2826378ed945.png)


5 - To **your question** number 9. It's correct. The weight of the selected runs anchor point values has a different weight and colour than the rest.


Many thanks

tynandebold commented:
@comym I think I've addressed all your concerns from your previous comment. Please have a look at let me know what you think!

comym commented:
Hi,

After checking Gitpod these are my current comments:

1 - When hovering over a run line, the plot highlights some random values across different metrics and positions. This should not happen. Only the related values to the on-hover run should show.

![Screenshot 2022-11-23 at 13 00 31](https://user-images.githubusercontent.com/55446576/203594277-06c95250-1b19-48d5-823f-11d852070bb5.png)


2 - Please double-check if the colour of the text values is the same as the line in the background when hovering over a run. They **should not** be the same. The different colours allow the text on top to be readable. The design shows the colours to be different as seen below. 

![Screenshot 2022-11-23 at 16 09 26](https://user-images.githubusercontent.com/55446576/203594792-8de82bd0-d701-49de-a48a-3a6a6dfb79e4.png)


3 - When hovering over a metric name, the current build only highlights the axis line and some values. This is right but this is also only one of the changes that should occur in this interaction. What we're missing here is to knock back (decrease in contrast) all the other metric columns that are not being highlighted, so that the highlighted one can pop to the front. Also in this state, we knock back many of the other lines and labels. Basically eliminating all the unnecessary noise at that point. See the design for reference.

![Screenshot 2022-11-23 at 16 11 39](https://user-images.githubusercontent.com/55446576/203595515-a2891514-4d64-44db-b7c0-500b818f0917.png)


4 - Uneven margins/padding. The current version has improved the big gap we had before, however it's still unbalanced. In the image below, if we consider the two margins/padding in blue to be the same, we can still see there's a big difference between the top and bottom - marked in red.  We should even it out. Please double-check the designs for reference.

![Screenshot 2022-11-23 at 16 15 21](https://user-images.githubusercontent.com/55446576/203596107-5f958961-394f-4d88-8aa8-f0afb8e06bbc.png)


5 - I've recently checked Gitpod again and I've seen some variations of it. Now showing only two metric columns. Not sure if that's working in progress but I'll comment anyway just in case.

Metric names are gigantic and label values have too many decimal values. What's happening is that they are being caped by some sort of mask or area limit as seen below. We should not show all these decimal values and shorten them somehow. I'm happy to talk about it and explore solutions if this is going to be a reality.

![Screenshot 2022-11-23 at 16 21 02](https://user-images.githubusercontent.com/55446576/203597346-aec484bd-28fc-4046-aef5-8b08d9dec26d.png)


6 - Overlapping labels. As marked in red below. There's a mix of very long labels and labels with fewer decimal values. These values are unreadable. I'm happy to talk about it and explore solutions if this is going to be a reality.

![Screenshot 2022-11-23 at 16 23 38](https://user-images.githubusercontent.com/55446576/203597924-c11c6fce-a1e9-48c6-b8d8-b4f26d0bd60c.png)


If you have any questions just reach out and we can talk about it in detail instead of writing long text blocks here.

Many thanks!
